### PR TITLE
cmd/scollector: add mssql.replica.resource_state metrics 

### PR DESCRIPTION
**Pull request by gbrayut**

### CODE REVIEW

File: cmd/scollector/collectors/sql_windows.go [hunk 1]
```diff
@@ -30,20 +30,33 @@ func init() {
 	c_replica_server.init = wmiInit(c_replica_server, func() interface{} { return &[]Win32_PerfRawData_MSSQLSERVER_SQLServerAvailabilityReplica{} }, `WHERE Name <> '_Total'`, &sqlAGQuery)
 	collectors = append(collectors, c_replica_server)
 
-	var hostname, _ = os.Hostname()
+	hostname, _ = os.Hostname()
```
Comment (mjibson) [inline cmd/scollector/collectors/sql_windows.go]: We can just switch to `util.Hostname` here. It gets set and correctly cleaned and is used in other parts of scollector.

File: cmd/scollector/collectors/sql_windows.go [hunk 2]
```diff
@@ -30,20 +30,33 @@ func init() {
...
+const (
+	rootMSCluster string = "root\\MSCluster"
+)
+
 var (
-	sqlQuery     string
-	sqlAGDBQuery string
-	sqlAGQuery   string
-	sqlAGVotes   string
+	hostname       string
```
Comment (mjibson) [inline cmd/scollector/collectors/sql_windows.go]: This can be removed.

File: cmd/scollector/collectors/sql_windows.go [hunk 3]
```diff
@@ -30,20 +30,33 @@ func init() {
...
 		F:        c_mssql_replica_votes,
 		Interval: time.Minute * 5,
 	}
-	c_replica_votes.init = wmiInitNamespace(c_replica_votes, func() interface{} { return &[]MSCluster_Node{} }, fmt.Sprintf("WHERE Name = '%s'", hostname), &sqlAGVotes, "Root\\MSCluster")
+	c_replica_votes.init = wmiInitNamespace(c_replica_votes, func() interface{} { return &[]MSCluster_Node{} }, fmt.Sprintf("WHERE Name = '%s'", hostname), &sqlAGVotes, rootMSCluster)
 	collectors = append(collectors, c_replica_votes)
+
+	c_replica_resources := &IntervalCollector{
+		F:        c_mssql_replica_resources,
+		Interval: time.Minute * 1,
```
Comment (mjibson) [inline cmd/scollector/collectors/sql_windows.go]: Remove the ` * 1`

File: cmd/scollector/collectors/sql_windows.go [hunk 4]
```diff
@@ -446,3 +459,29 @@ type MSCluster_Node struct {
 	DynamicWeight uint32
 	State         uint32
 }
+
+func c_mssql_replica_resources() (opentsdb.MultiDataPoint, error) {
+	var dst []MSCluster_Resource
+	//Only report metrics for resources owned by this node
+	var q = wmi.CreateQuery(&dst, fmt.Sprintf("WHERE OwnerNode = '%s'", hostname))
```
Comment (mjibson) [inline cmd/scollector/collectors/sql_windows.go]: util.Hostname

**Comment by gbrayut** [comment]
Update with fixes

**Pull request merged**